nixos/modules/services/security/crowdsec.nix

```diff
@@ -489,182 +398,189 @@ in
-        ---
-        ${lib.strings.concatMapStringsSep "\n---\n" builtins.toJSON cfg.localConfig.acquisitions}
-        ---
+        $sudo ${lib.getExe' cfg.package "cscli"} -c=${"${cfg.settings.general.config_paths.config_dir}/config.yaml"} "$@"
       '';
 
       scriptArray = [
-        "set -euo pipefail"
-        "${lib.getExe' pkgs.coreutils "mkdir"} -p '${hubDir}'"
         "${lib.getExe cscli} hub update"
       ]
-      ++ lib.optionals (cfg.hub.collections != [ ]) [
-        "${lib.getExe cscli} collections install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.collections
-        }"
-      ]
-      ++ lib.optionals (cfg.hub.scenarios != [ ]) [
-        "${lib.getExe cscli} scenarios install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.scenarios
-        }"
-      ]
-      ++ lib.optionals (cfg.hub.parsers != [ ]) [
-        "${lib.getExe cscli} parsers install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.parsers
-        }"
-      ]
-      ++ lib.optionals (cfg.hub.postOverflows != [ ]) [
-        "${lib.getExe cscli} postoverflows install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.postOverflows
-        }"
-      ]
-      ++ lib.optionals (cfg.hub.appSecConfigs != [ ]) [
-        "${lib.getExe cscli} appsec-configs install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.appSecConfigs
-        }"
-      ]
-      ++ lib.optionals (cfg.hub.appSecRules != [ ]) [
-        "${lib.getExe cscli} appsec-rules install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.appSecRules
-        }"
-      ]
-      ++ lib.optionals (cfg.settings.general.api.server.enable) [
-        ''
-          if [ ! -s "${cfg.settings.general.api.client.credentials_path}" ]; then
-            ${lib.getExe cscli} machine add "${cfg.name}" --auto
-          fi
-        ''
-      ]
-      ++ lib.optionals (cfg.settings.capi.credentialsFile != null) [
-        ''
-          if ! ${lib.getExe pkgs.gnugrep} -q password "${cfg.settings.capi.credentialsFile}" ]; then
-            ${lib.getExe cscli} capi register
-          fi
-        ''
-      ]
-      ++ lib.optionals (cfg.settings.console.tokenFile != null) [
-        ''
-          if [ ! -e "${cfg.settings.console.tokenFile}" ]; then
-            ${lib.getExe cscli} console enroll "$(cat ${cfg.settings.console.tokenFile})" --name ${cfg.name}
-          fi
-        ''
-      ];
+      ++ lib.optional cfg.settings.general.api.server.enable ''
+        if [ ! -f ${cfg.settings.general.api.client.credentials_path} ]; then
+          echo "No local API credentials currently created. Generating local API credentials..."
+          ${lib.getExe cscli} machines add nixos-generated --auto --file ${cfg.settings.general.api.client.credentials_path}
```

Comment:
Was there a reason the hostname was changed to `nixos-generated`? It was `"${cfg.name}"` earlier.